analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConstFix.cs

```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[ExportCodeFixProvider(LanguageNames.CSharp)]
+public sealed class UnchangedLocalVariablesShouldBeConstFix : SonarCodeFix
+{
+    private const string Title = "Convert to constant.";
+
+    public override ImmutableArray<string> FixableDiagnosticIds =>
+        ImmutableArray.Create(UnchangedLocalVariablesShouldBeConst.DiagnosticId);
+
+    protected override async Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext context)
+    {
+        var oldNode = VariableDeclaration(root, context);
+        var declaration = oldNode.Type.IsVar
+            ? WithExplictType(oldNode, await context.Document.GetSemanticModelAsync())
+            : oldNode;
+        var newNode = root.ReplaceNode(oldNode, ConstantDeclaration(declaration));
+
+        context.RegisterCodeFix(
+            CodeAction.Create(
+            Title,
+            token => Task.FromResult(context.Document.WithSyntaxRoot(newNode))),
+            context.Diagnostics);
+    }
+    private static VariableDeclarationSyntax VariableDeclaration(SyntaxNode root, CodeFixContext context) =>
+        (VariableDeclarationSyntax)root.FindNode(context.Diagnostics.First().Location.SourceSpan).Parent;
```

Comment:
Let's play it safe:
```suggestion
        root.FindNode(context.Diagnostics.First().Location.SourceSpan)?.Parent as VariableDeclarationSyntax;
```

Reply:
It is a bit more complicated than this. There might be multiple declarators:

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/statements#1262-local-variable-declarations

```cs
int i1 = 1, i2 = 2;
```

I didn't look up what the rule does in this case, but you should either

* Support multiple declarators or
* Don't offer a fix if multiple declarators are present

You also need to add a test case to *ToFix* in any case (I made a suggestion).